lib/engine/step/tokener.rb

```diff
@@ -55,6 +55,11 @@ def place_token(entity, city, token, connected: true, extra_action: false,
                            "#{special_ability.city}, not on city #{city.index}"
         end
 
+        if special_ability&.type == :token && special_ability.hexes.any? && !special_ability.hexes.include?(hex.id)
```

Comment:
Prefer `!empty?` over `any?`. The latter is much slower due to the way it is implemented in Opal.

Reply:
ah, I was copypasting from https://github.com/benjaminxscott/18xx/blob/master/lib/engine/step/tokener.rb#L132

Reply:
Its wrong there too then :-D